source/extensions/filters/common/ext_authz/ext_authz_http_impl.cc

```diff
@@ -0,0 +1,117 @@
+#include "extensions/filters/common/ext_authz/ext_authz_http_impl.h"
+
+#include "common/common/enum_to_int.h"
+#include "common/http/async_client_impl.h"
+
+#include "absl/strings/str_cat.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Filters {
+namespace Common {
+namespace ExtAuthz {
+
+namespace {
+
+const Http::HeaderMap* getZeroContentLengthHeader() {
+  static const Http::HeaderMap* header_map =
+      new Http::HeaderMapImpl{{Http::Headers::get().ContentLength, std::to_string(0)}};
+  return header_map;
+}
+
+} // namespace
+
+RawHttpClientImpl::RawHttpClientImpl(
+    const std::string& cluster_name, Upstream::ClusterManager& cluster_manager,
+    const absl::optional<std::chrono::milliseconds>& timeout, const std::string& path_prefix,
+    const std::vector<Http::LowerCaseString>& response_headers_to_remove)
+    : cluster_name_(cluster_name), path_prefix_(path_prefix),
+      response_headers_to_remove_(response_headers_to_remove), timeout_(timeout),
+      cm_(cluster_manager) {}
+
+RawHttpClientImpl::~RawHttpClientImpl() { ASSERT(!callbacks_); }
+
+void RawHttpClientImpl::cancel() {
+  ASSERT(callbacks_ != nullptr);
+  request_->cancel();
+  callbacks_ = nullptr;
+}
+
+void RawHttpClientImpl::check(RequestCallbacks& callbacks,
+                              const envoy::service::auth::v2alpha::CheckRequest& request,
+                              Tracing::Span&) {
+  ASSERT(callbacks_ == nullptr);
+  callbacks_ = &callbacks;
+
+  Http::HeaderMapPtr headers = std::make_unique<Http::HeaderMapImpl>(*getZeroContentLengthHeader());
+  for (const auto& header : request.attributes().request().http().headers()) {
+    const Http::LowerCaseString key{header.first};
+    if (key == Http::Headers::get().Path && !path_prefix_.empty()) {
```

Comment:
There's some missing coverage in this file: https://64274-65214191-gh.circle-artifacts.com/0/build/envoy/generated/coverage/coverage.source_extensions_filters_common_ext_authz_ext_authz_http_impl.cc.html